src/Controllers/CourseController.php

```diff
@@ -18,6 +18,36 @@
 
 class CourseController extends Controller
 {
+    public function view()
+    {
+        try {
+            $user = auth()->user();
+
+            $courseUserIds = CourseUser::where(
+                'user_id', '=', $user->id
+            )->pluck('course_id')->toArray();
+            $courses = [];
+
+            foreach ($courseUserIds as $courseUserId) {
+                $course = Course::find($courseUserId);
+                array_push($courses, $course);
+            }
```

Comment:
Have you tried to join the query directly? Calling each `Course` model like this will result in multiple db queries, e.g.
```sql
SELECT * FROM course_user WHERE id=<id1>;
SELECT * FROM course_user WHERE id=<id2>;
SELECT * FROM course_user WHERE id=<id3>;
...
```
and the cost will be quite expensive

Reply:
i tried to fix this, please praise me if you think it's good enough uwu